src/pages/Search/SearchTransactionsChangeReport.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {useMemo} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import {useSearchContext} from '@components/Search/SearchContext';
+import type {ListItem} from '@components/SelectionList/types';
+import {changeTransactionsReport} from '@libs/actions/Transaction';
+import Navigation from '@libs/Navigation/Navigation';
+import IOURequestEditReportCommon from '@pages/iou/request/step/IOURequestEditReportCommon';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+
+type ReportListItem = ListItem & {
+    /** reportID of the report */
+    value: string;
+};
+
+function SearchTransactionsChangeReport() {
+    const {selectedTransactions, clearSelectedTransactions} = useSearchContext();
+    const selectedTransactionsKeys = useMemo(() => Object.keys(selectedTransactions), [selectedTransactions]);
+
+    const [allReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {canBeMissing: false});
+    const transactionsReports = useMemo(() => {
+        const reports = Object.values(selectedTransactions).reduce((acc, transaction) => {
+            const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${transaction.reportID}`];
+            if (report) {
+                acc.add(report);
+            }
+            return acc;
+        }, new Set<Report>());
+        return [...reports];
+    }, [allReports, selectedTransactions]);
+
+    const selectReport = (item: ReportListItem) => {
```

Comment:
Hi! This is causing https://github.com/Expensify/App/issues/62500 - we need to add similar logic to [this](https://github.com/Expensify/App/blob/a45da7e6b375a7b2d9896d17b8e40a49fb3ad5b5/src/pages/iou/request/step/IOURequestStepReport.tsx#L33-L38) so that it doesn't try to move the report if it's being moved to the report it's already on.